src/pkg/backup/details/entry.go

```diff
@@ -68,32 +71,48 @@ func (de Entry) ToLocationIDer(backupVersion int) (LocationIDer, error) {
 		return de.ItemInfo.uniqueLocation(baseLoc)
 	}
 
-	if backupVersion >= version.OneDrive7LocationRef ||
-		(de.ItemInfo.infoType() != OneDriveItem &&
-			de.ItemInfo.infoType() != SharePointLibrary) {
-		return nil, clues.New("no previous location for entry")
-	}
-
-	// This is a little hacky, but we only want to try to extract the old
-	// location if it's OneDrive or SharePoint libraries and it's known to
-	// be an older backup version.
-	//
-	// TODO(ashmrtn): Remove this code once OneDrive/SharePoint libraries
-	// LocationRef code has been out long enough that all delta tokens for
-	// previous backup versions will have expired. At that point, either
-	// we'll do a full backup (token expired, no newer backups) or have a
-	// backup of a higher version with the information we need.
 	rr, err := path.FromDataLayerPath(de.RepoRef, true)
 	if err != nil {
-		return nil, clues.Wrap(err, "getting item RepoRef")
+		return nil, clues.Wrap(err, "getting item RepoRef").
+			With("repo_ref", de.RepoRef)
 	}
 
-	p, err := path.ToDrivePath(rr)
-	if err != nil {
-		return nil, clues.New("converting RepoRef to drive path")
+	var baseLoc *path.Builder
+
+	switch de.ItemInfo.infoType() {
+	case ExchangeEvent:
+		if backupVersion >= 2 {
+			return nil, clues.New("no previous location for calendar entry").
+				With("repo_ref", rr)
+		}
+
+		fallthrough
+	case ExchangeMail:
+		fallthrough
+	case ExchangeContact:
```

Comment:
```suggestion
	case ExchangeMail, ExchangeContact:
```

Reply:
ahhh I always forget you can do that for some reason haha